server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -26,136 +26,251 @@
         }
 
         /**
          * Register a ValuesSource to Aggregator mapping.  This version provides a convenience method for mappings that only apply to a
          * known list of {@link ValuesSourceType}, to allow passing in the type and auto-wrapping it in a predicate
-         *  @param aggregationName The name of the family of aggregations, typically found via
-         *                         {@link ValuesSourceAggregationBuilder#getType()}
-         * @param valuesSourceTypes The ValuesSourceTypes this mapping applies to.
+         *
+         * @param aggregationName    The name of the family of aggregations, typically found via
+         *                           {@link ValuesSourceAggregationBuilder#getType()}
+         * @param valuesSourceTypes  The ValuesSourceTypes this mapping applies to.
          * @param aggregatorSupplier An Aggregation-specific specialization of AggregatorSupplier which will construct the mapped aggregator
          *                           from the aggregation standard set of parameters
          */
         public void register(String aggregationName, List<ValuesSourceType> valuesSourceTypes, AggregatorSupplier aggregatorSupplier) {
-            register(aggregationName, (candidate) -> {
-                for (ValuesSourceType valuesSourceType : valuesSourceTypes) {
-                    if (valuesSourceType.equals(candidate)) {
-                        return true;
-                    }
-                }
-                return false;
-            }, aggregatorSupplier);
+            register(aggregationName, new MultiRegistryEntry(aggregatorSupplier, valuesSourceTypes));
         }
 
         /**
          * Register an aggregator that applies to any values source type.  This is a convenience method for aggregations that do not care at
          * all about the types of their inputs.  Aggregations using this version of registration should not make any other registrations, as
          * the aggregator registered using this function will be applied in all cases.
          *
-         * @param aggregationName The name of the family of aggregations, typically found via
-         *                        {@link ValuesSourceAggregationBuilder#getType()}
+         * @param aggregationName    The name of the family of aggregations, typically found via
+         *                           {@link ValuesSourceAggregationBuilder#getType()}
          * @param aggregatorSupplier An Aggregation-specific specialization of AggregatorSupplier which will construct the mapped aggregator
          *                           from the aggregation standard set of parameters.
          */
         public void registerAny(String aggregationName, AggregatorSupplier aggregatorSupplier) {
-            register(aggregationName, (ignored) -> true, aggregatorSupplier);
+            register(aggregationName, new AnyRegistryEntry(aggregatorSupplier));
```

Comment:
So, @csoulios is working on a near term plan to get rid of the literal any, and replace registerAny with just registering all the `CoreValuesSourceTypes`.  We could adopt something like that here, and just register an explicit list of `SingleRegistryEntry` cases instead.  Or leave this as is and change it over when the rest of Christos's ANY work merges.

Reply:
Just pushed PR #55747 removing `registerAny()` method